src/main/java/frc/robot/subsystems/VisionControl.java

```diff
@@ -272,17 +265,48 @@ private double calculateBallRotation() {
         return -ball_rotation;
     }
 
-    public double calculateShooterRPMS() {
+    public void autoShoot() {
+        switch (shooterstate) {
+            case ALIGN:
+                trackHoop();
+                if (Math.abs(hoopr) <= hoopChassisThreshold) {
+                    shooterstate = shooterState.WAIT;
+                }
+                break;
+            case WAIT:
+                double rpm = calculateShooterRPMS();
+                shooter.startShoter(rpm);
+                if (Math.abs(shooter.getShooterRpms() - rpm) < shooterThreshold) {
+                    shooterstate = shooterState.SHOOT;
+                }
+                break;
+            case SHOOT:
+                double rpms = calculateShooterRPMS();
+                shooter.startShoter(rpms);
+                shooter.startFeeder();
+                if (Math.abs(shooter.getShooterRpms() - rpms) > shooterThreshold) {
+                    shooter.stopFeeder();
+                    shooterstate = shooterState.WAIT;
+                }
+                break;
+        }
+    }
+
+    public boolean isHoopValid() {
+        return visionData.isHoopValid();
+    }
+
+    private double calculateShooterRPMS() {
```

Comment:
We need to fix this once we test and get the calibration curve

Reply:
yep